lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -383,12 +383,25 @@ def add_pods_library
               frameworks.remove_reference(old_product_ref)
             end
 
+            product_name = target.product_name
+
+            # Delete previously integrated references.
+            product_build_files = build_phase.files.select do |build_file|
+              # build_file.display_name =~ Pod::Deintegrator::FRAMEWORK_NAMES
+              build_file.display_name =~ /^(libPods.*\.a)|(Pods.*\.framework)$/i
+            end
+
+            product_build_files.each do |product_file|
+              next unless product_name != product_file.display_name
+              build_phase.remove_build_file(product_file)
+            end
+
             # Find or create and add a reference for the current product type
             target_basename = target.product_basename
```

Comment:
this variable is now a dupe of `product_name` declared above

Reply:
it is not? 
```ruby
product_name = target.product_name
```
vs
```ruby
target_basename = target.product_basename
```

Reply:
ah yeah realized that from your above comment, I see